qiskit_ibm/ibm_provider.py

```diff
@@ -178,53 +152,61 @@ def __new__(
             IBMProviderCredentialsInvalidToken: If the `token` is not a valid
                 IBM Quantum token.
         """
-        account_credentials, account_preferences, hgp = cls._resolve_credentials(
+        # pylint: disable=unused-argument,unsubscriptable-object
+        super().__init__()
+        account_credentials, account_preferences = self._resolve_credentials(
             token=token,
             url=url,
-            hub=hub,
-            group=group,
-            project=project,
             **kwargs
         )
-        hub, group, project = hgp.to_tuple()
-        if not cls._credentials and (not cls._providers or
-                                     cls._is_different_account(account_credentials.token)):
-            cls._initialize_providers(credentials=account_credentials,
-                                      preferences=account_preferences)
-            return cls._get_provider(hub=hub, group=group, project=project)
-        elif not cls._credentials and cls._providers:
-            return cls._get_provider(hub=hub, group=group, project=project)
-        else:
-            return object.__new__(cls)
+        self._initialize_backend_groups(credentials=account_credentials,
+                                        preferences=account_preferences)
+        if self._backend_groups and self._backend_group:
+            self.credentials = self._backend_group.credentials
+            self._api_client = self._backend_group._api_client
+            # Initialize the internal list of backends.
+            self._backends: Dict[str, IBMBackend] = {}
+            # Open access backend group is always first on list, overwrite with backends from
+            # premium backend groups since they take higher priority
+            for backend_group in self._backend_groups.values():
+                self._backends.update(backend_group.backends)
+            # Then overwrite with backends from resolved backend group
+            self._backends.update(self._backend_group.backends)
+            # Initialize backend service
+            self._backend = IBMBackendService(self)
+            # Initialize other services.
+            self._random = IBMRandomService(self) \
+                if self.credentials.extractor_url else None
+            self._experiment = IBMExperimentService(self) \
+                if self.credentials.experiment_url else None
+            self._runtime = IBMRuntimeService(self) \
+                if self.credentials.runtime_url else None
```

Comment:
Say if I have 3 hgp, only hgp2 has access to runtime. But unluckily `_initialize_backend_groups()` picked hgp3 as the "backend group" to use. Now all of a sudden I can't access runtime anymore? You probably want to search all "backend groups" until you find an url before setting it to None (which can be done when you're looping through all hgp, btw).